tree/treeplayer/inc/ROOT/TVec.hxx

```diff
@@ -0,0 +1,535 @@
+}
+
+template <typename T, typename V>
+auto operator%(const TVec<T> &v0, const TVec<V> &v1) -> TVec<decltype(v0[0] % v1[0])>
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "%", [](const T &t, const V &v) { return t % v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator>(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, ">", [](const T &t, const V &v) -> Boolean_t { return t > v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator>=(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, ">=", [](const T &t, const V &v) -> Boolean_t { return t >= v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator==(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "==", [](const T &t, const V &v) -> Boolean_t { return t == v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator!=(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "!=", [](const T &t, const V &v) -> Boolean_t { return t != v; });
+}
+template <typename T, typename V>
+TVec<Boolean_t> operator<=(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "<=", [](const T &t, const V &v) -> Boolean_t { return t <= v; });
+}
+
+template <typename T, typename V>
+TVec<Boolean_t> operator<(const TVec<T> &v0, const TVec<V> &v1)
+{
+   return ROOT::Internal::VecOps::Operate(v0, v1, "<", [](const T &t, const V &v) -> Boolean_t { return t < v; });
+}
+///@}
+
+/** @name Math Functions
+ *  Math functions on TVecs
+*/
+///@{
+#define MATH_FUNC(FUNCNAME)                                                                   \
+   template <typename T>                                                                      \
+   auto FUNCNAME(const TVec<T> &v)->TVec<decltype(std::FUNCNAME(v[0]))>                       \
+   {                                                                                          \
+      return ROOT::Internal::VecOps::Operate(v, [](const T &t) { return std::FUNCNAME(t); }); \
+   }
+
+MATH_FUNC(sqrt)
+MATH_FUNC(log)
+MATH_FUNC(sin)
+MATH_FUNC(cos)
+MATH_FUNC(asin)
+MATH_FUNC(acos)
+MATH_FUNC(tan)
+MATH_FUNC(atan)
+MATH_FUNC(sinh)
+MATH_FUNC(cosh)
+MATH_FUNC(asinh)
+MATH_FUNC(acosh)
+MATH_FUNC(tanh)
+MATH_FUNC(atanh)
+
+///@}
+
+/// Inner product
+template <typename T, typename V>
+auto Dot(const TVec<T> v0, const TVec<V> v1) -> decltype(v0[0] * v1[0])
+{
+   ROOT::Internal::VecOps::CheckSizes(v0.size(), v1.size(), "Dot");
+   return std::inner_product(v0.begin(), v0.end(), v1.begin(), decltype(v0[0] * v1[0])(0));
+}
+
+/// Sum elements
+template <typename T>
+T Sum(const TVec<T> v)
+{
+   return std::accumulate(v.begin(), v.end(), 0);
+}
+
+template <typename T, typename F>
+TVec<typename TCallTraits<F>::ret_type> Map(const TVec<T> &v, F f)
+{
+   return ROOT::Internal::VecOps::Operate(v, f);
+}
+
```

Comment:
I was aggressively removing entities which could be expressed otherwise. Given the cost, we can keep the aforementioned helper instead of a syntax like v[Map(v, unaryOp)]...

Reply:
Addressed.